### PR TITLE
chore: add TLS by default for cloud deployments

### PR DESCRIPTION
**Pull request by stoksc**
We should, in cloud setups, setup TLS for communication with
the postgres database, since credentials and models are
sent between it.

# Test Plan
- [x] run terraform and verify we connect to db
- [x] run cloudformation (make upstream branch)

### CODE REVIEW

**Comment by stoksc** [comment]
title is misleading, it's specifically not by default 🤦 only when you use the secure cloud formation or add TF_VAR__SSL_ENABLED or whatever on gcp. I can edit that msg.

File: master/internal/db/config.go [hunk 1]
```diff
@@ -4,15 +4,18 @@ package db
 func DefaultConfig() *Config {
 	return &Config{
 		Migrations: "file://static/migrations",
+		SSLMode:    sslModeDisable,
 	}
 }
 
 // Config hosts configuration fields of the database.
 type Config struct {
```
Comment (stoksc) [inline master/internal/db/config.go:12]: I'm sortof partial to removing all this and just having `connection_string` as opposed to eventually supporting ever pg option

File: master/internal/db/setup.go [hunk 2]
```diff
@@ -9,15 +9,20 @@ import (
...
 
 // Setup connects to the database and run any necessary migrations.
 func Setup(opts *Config) (*PgDB, error) {
 	dbURL := fmt.Sprintf(cnxTpl, opts.User, opts.Password, opts.Host, opts.Port, opts.Name)
+	dbURL += fmt.Sprintf(sslTpl, opts.SSLMode, opts.SSLRootCert)
 	log.Infof("connecting to database %s:%s", opts.Host, opts.Port)
 	db, err := ConnectPostgres(dbURL)
 	if err != nil {
-		return nil, errors.Wrapf(err, "connecting to database: %s", dbURL)
+		return nil, errors.Wrapf(err, "connecting to database: %s:%s", opts.Host, opts.Port)
```
Comment (stoksc) [inline master/internal/db/setup.go:25]: potentially on connection failures, we were printing the password 🙃

Comment (brain-good) [reply master/internal/db/setup.go:25]: 🙃

**Comment by brain-good** [comment]
@stoksc you'll need to go in and approve the jobs for the deploy and GPU tests 👍

**Comment by stoksc** [comment]
@brain-good wanted to get your opinions sort of on why this isn't working.. so basically, I followed
https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/UsingWithRDS.SSL.html which simply says, connect with sslmode=verify-ca (or verify-full) and sslrootcert=$CERT, where that's the cert they provide in the first paragraph. Doing that we get something like SSL Cert signed by unknown authority - so it can't be traced back to the root ca we provide libpq. 

Debugging this a bit, I expected the aurora rds to respond with some cert with `CN=*.cluster-csrkoc1nkoog.us-west-2.rds.amazonaws.com` (the database) issued by something like `CN=Amazon RDS us-west-2 2019 CA` with an intermediate for chaining the `CN=Amazon RDS us-west-2 2019 CA` cert to `Amazon RDS Root 2019 CA` and `Amazon RDS Root 2019 CA` is our `sslrootcert`.

What I see is a cert with `CN=*.cluster-csrkoc1nkoog.us-west-2.rds.amazonaws.com` issued by `OU=Server CA 1B, CN=Amazon` and an intermediate from `CN=Amazon,OU=Server CA 1B` to `Amazon Root CA 1` - amazon's root cert on amazontrust.com from https://www.amazontrust.com/repository/AmazonRootCA1.pem. Also a few more intermediates from other known amazon root ca's. So `psql "postgres://.../determined?sslmode=verify-ca&sslrootcert=AmazonRootCA1.pem"` works just fine. 🤷 With the other cert we hit https://golang.org/src/crypto/x509/verify.go?s=5144:5448#L880 being unable to complete a chain to `CN=Amazon RDS us-west-2 2019 CA`. I'm a little confused, tried to use the aws rds forum but I keep getting told I'm not allowed (by some automated thing).

Any thoughts?

**Comment by stoksc** [comment]
Got past gpu tests (which deployed a cluster) on upstream branch 👍

**Comment by stoksc** [comment]
Also, tested on GCP:
```
bradleylaney@det-master-brad-0-12-4:~$ sudo docker run     
    --name determined-master    
    --network determined     
    --restart unless-stopped     
    -p 8080:8080     
    -v /usr/local/determined/etc/master.yaml:/etc/determined/master.yaml     
    -v /usr/local/determined/etc/db_ssl_root_cert.pem:/etc/determined/etc/db_ssl_root_cert.pem     determinedai/determined-master:0.12.4.dev0
INFO[2020-05-14T17:45:37Z] master configuration: {"config_file":"","log":{"level":"info","color":true},"db":{"user":"postgres","password":"********","migrations":"file:///usr/share/determined/master/static/migrations","host":"10.105.0.3","port":"5432","name":"det-db-brad","ssl_mode":"verify-ca","ssl_root_cert":"/etc/determined/etc/db_ssl_root_cert.pem"},"scheduler":{"type":"fair_share","fit":"best"},"provisioner":{"agent_docker_image":"determinedai/determined-agent:0.12.4","agent_docker_network":"host","agent_docker_runtime":"runc","base_config":{"minCpuPlatform":"Intel Broadwell"},"boot_disk_size":200,"boot_disk_source_image":"projects/determined-ai/global/images/pedl-environments-e6662e6","container_startup_script":"","instance_type":{"gpu_num":8,"gpu_type":"nvidia-tesla-k80","machine_type":"n1-standard-32","preemptible":false},"label_key":"managed-by","label_value":"","master_url":"http://internal-ip:8080","max_idle_agent_period":"10m0s","max_instances":5,"name_prefix":"det-dynamic-agent-brad-","network_interface":{"external_ip":true,"network":"projects/determined-ai/global/networks/det-default-brad","subnetwork":"projects/determined-ai/regions/us-west1/subnetworks/det-default-brad"},"network_tags":["det-internal-det-default-brad","det-ssh-det-default-brad"],"project":"","provider":"gcp","service_account":{"email":"det-brad@determined-ai.iam.gserviceaccount.com","scopes":["https://www.googleapis.com/auth/cloud-platform"]},"startup_script":"","zone":""},"security":{"default_task":{"id":0,"user_id":0,"user":"root","uid":0,"group":"root","gid":0},"tls":{"cert":"","key":""},"http":true},"checkpoint_storage":{"bucket":"det-checkpoints-brad-816838","save_experiment_best":0,"save_trial_best":1,"save_trial_latest":1,"type":"gcs"},"task_container_defaults":{"shm_size_bytes":4294967296,"network_mode":"bridge"},"http_port":8080,"https_port":8443,"harness_path":"/opt/determined","root":"/usr/share/determined/master","telemetry":{"enabled":true,"segment_master_key":"********","segment_webui_key":"********"}} 
INFO[2020-05-14T17:45:37Z] Determined master 0.12.4.dev0 (built with go1.13) 
INFO[2020-05-14T17:45:37Z] connecting to database 10.105.0.3:5432       
INFO[2020-05-14T17:45:37Z] running migrations from file:///usr/share/determined/master/static/migrations 
INFO[2020-05-14T17:45:37Z] unable to find golang-migrate version        
INFO[2020-05-14T17:45:37Z] deleting all searcher events for terminal state experiments 
INFO[2020-05-14T17:45:37Z] found provisioner configuration              
INFO[2020-05-14T17:45:37Z] connecting to GCP cluster                    
INFO[2020-05-14T17:45:37Z] telemetry reporting is enabled; run with `--telemetry-enabled=false` to disable 
⇨ http server started on [::]:8080
INFO[2020-05-14T17:45:42Z] found 0 instances:                            id=provisioner system=master type=Provisioner
```

**Pull request merged**